core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/provider.go

```diff
@@ -384,7 +384,8 @@ func (p *logEventProvider) readLogs(ctx context.Context, latest int64, filters [
 		if configUpdateBlock := int64(filter.configUpdateBlock); start < configUpdateBlock {
 			start = configUpdateBlock
 		}
-		logs, err := p.poller.LogsWithSigs(start, latest, filter.topics, common.BytesToAddress(filter.addr), pg.WithParentCtx(ctx))
+		// query logs based on contract address, event sig, and blocks
+		logs, err := p.poller.LogsWithSigs(start, latest, []common.Hash{filter.topics[0]}, common.BytesToAddress(filter.addr), pg.WithParentCtx(ctx))
```

Comment:
Did we consider adding a new query to the log poller? I think we could add a very simple function to log poller to suite our needs here and then we can avoid filtering on the node side. Something like...

```go
func LogsInRangeWithTopics(start, end int64, address common.Address, topics []common.Hash, qopts ...pg.QOpt) ([]Log, error) 
```

would need a little tweaking to allow for "any" matchers in the topics. Maybe topics could be `[]*common.Hash`.